linux/ibus-keyman/src/engine.c

```diff
@@ -459,87 +459,84 @@ ibus_keyman_engine_destroy (IBusKeymanEngine *keyman)
-    KM_KBP_VKEY_J,    //        public static final int KEY_J = 0x24;
-    KM_KBP_VKEY_K,    //        public static final int KEY_K = 0x25;
-    KM_KBP_VKEY_L,    //        public static final int KEY_L = 0x26;
-    KM_KBP_VKEY_COLON,    //        public static final int KEY_SEMICOLON = 0x27;
-    KM_KBP_VKEY_QUOTE,    //        public static final int KEY_APOSTROPHE = 0x28;
-    KM_KBP_VKEY_BKQUOTE,    //        public static final int KEY_GRAVE = 0x29;
-    0,      //        public static final int KEY_LEFTSHIFT = 0x2A;
-    KM_KBP_VKEY_BKSLASH,    //        public static final int KEY_BACKSLASH = 0x2B;
-    KM_KBP_VKEY_Z,    //        public static final int KEY_Z = 0x2C;
-    KM_KBP_VKEY_X,    //        public static final int KEY_X = 0x2D;
-    KM_KBP_VKEY_C,    //        public static final int KEY_C = 0x2E;
-    KM_KBP_VKEY_V,    //        public static final int KEY_V = 0x2F;
-    KM_KBP_VKEY_B,    //        public static final int KEY_B = 0x30;
-    KM_KBP_VKEY_N,    //        public static final int KEY_N = 0x31;
-    KM_KBP_VKEY_M,    //        public static final int KEY_M = 0x32;
-    KM_KBP_VKEY_COMMA,    //        public static final int KEY_COMMA = 0x33;
-    KM_KBP_VKEY_PERIOD,    //        public static final int KEY_DOT = 0x34;
-    KM_KBP_VKEY_SLASH,    //        public static final int KEY_SLASH = 0x35;
-    0,      //        public static final int KEY_RIGHTSHIFT = 0x36;
-    KM_KBP_VKEY_NPSTAR,      //        public static final int KEY_KPASTERISK = 0x37;
-    0,      //        public static final int KEY_LEFTALT = 0x38;
-    KM_KBP_VKEY_SPACE,     //        public static final int KEY_SPACE = 0x39;
-    0,      //        public static final int KEY_CAPSLOCK = 0x3A;
+    0,                   //        padding = 0x00;
+    KM_KBP_VKEY_ESC,     //        public static final int KEY_ESC = 0x01;
+    KM_KBP_VKEY_1,       //        public static final int KEY_1 = 0x02;
+    KM_KBP_VKEY_2,       //        public static final int KEY_2 = 0x03;
+    KM_KBP_VKEY_3,       //        public static final int KEY_3 = 0x04;
+    KM_KBP_VKEY_4,       //        public static final int KEY_4 = 0x05;
+    KM_KBP_VKEY_5,       //        public static final int KEY_5 = 0x06;
+    KM_KBP_VKEY_6,       //        public static final int KEY_6 = 0x07;
+    KM_KBP_VKEY_7,       //        public static final int KEY_7 = 0x08;
+    KM_KBP_VKEY_8,       //        public static final int KEY_8 = 0x09;
+    KM_KBP_VKEY_9,       //        public static final int KEY_9 = 0x0A;
+    KM_KBP_VKEY_0,       //        public static final int KEY_0 = 0x0B;
+    KM_KBP_VKEY_HYPHEN,  //        public static final int KEY_MINUS = 0x0C;
+    KM_KBP_VKEY_EQUAL,   //        public static final int KEY_EQUALS = 0x0D;
+    KM_KBP_VKEY_BKSP,    //        public static final int KEY_BACKSPACE = 0x0E;
+    KM_KBP_VKEY_TAB,     //        public static final int KEY_TAB = 0x0F;
+    KM_KBP_VKEY_Q,       //        public static final int V_R = 0x13;
```

Comment:
Did you intend to remove keys <kbd>W</kbd>, <kbd>E</kbd>, and <kbd>R</kbd>?

Reply:
ehhh - no. But who needs <kbd>W</kbd>, <kbd>E</kbd>, and <kbd>R</kbd> anyways :smile: 

Thanks!